tools/codegen/codespec/element_type.go

```diff
@@ -0,0 +1,31 @@
+package codespec
+
+import "fmt"
+
+type ElemType int
+
+const (
+	Bool ElemType = iota
+	Float64
+	Int64
+	Number
+	String
+	Unknown
+)
```

Comment:
Would leave the model together in `model.go`, and having specific logic for api spec -> model in a separate file. In terms of the unknown might be better to error out as we would be able to generate a "correct" model.

Reply:
For Unknown, it does throw an error except if the element is an array or object in which case it is essentially silently ignored. Unknown is more like a place holder for nil ElemType.
How would you want to handle this scenario from the schema generation side? i.e. if the elem type is an object or array?

Reply:
From code generation side it currently support existing model element types (Bool, Float64, Int64, Number, String), which brought me the question of what to do in case of an unknown.
SInce this is currently a limitation, I see 2 options:
- Best effort and simply default to a string. This would not be a correct generation can still be helpful as when can see generation of all the rest of the schema.
- Error out directly, more transparent but we wont get value of a potential "partial generation".
More inclined towards first option.

Reply:
I'm more inclined towards the first as well. Would like this to be best effort instead of completely breaking for unhandled cases. We can always continue improving as we come across more and more test cases.

Let me update to default to string in that case.

Reply:
updated